code/modules/mob/living/basic/lavaland/legion/legion.dm

```diff
@@ -72,6 +72,7 @@
 	consumed.extinguish_mob()
 	consumed.fully_heal(HEAL_DAMAGE)
 	consumed.apply_status_effect(/datum/status_effect/grouped/stasis, STASIS_LEGION_EATEN)
+	RegisterSignal(consumed, COMSIG_LIVING_REVIVE, PROC_REF(on_consumed_revive))
```

Comment:
Unregister this on exit, _just in case_ it is removed via an unusual means (an admin might get it out using verbs, or there is probably some means of teleportation which can retrieve them)